code/game/over_and_under_lays.dm

```diff
@@ -0,0 +1,91 @@
+//The deets:
+//To use, edit one of the 2 sublists in overlay_list/underlay_list then call redraw_overlays or redraw_underlays
+//The sublists are:
+//	(overlay used as example, underlay contains the same layers, only replace overlay with underlay)
+//	OVERLAY_APPEARANCE - overlay is part of the atom, and should be concidered to be defining the atom's appearance
+//		(think limbs on a mob)
+//
+//	OVERLAY_PRIORITY
+//	UNDERLAY_PRIORITY
+//		Mainly for outside use. Things that edit other thing's overlays or underlays can use these
+//		to keep them from getting cleared by that thing. for overlays they appear above the other overlays
+//		for underlays they appear below the other underlays
+//
+//	clear_overlays() does not reset priority overlays, the thing that added the overlays is
+//		responsable for removing the individual overlays it added when needed.
+//
+//	clear_overlays() does not call redraw_overlays() or redraw them, you must call redraw_overlays.
+//		(This is so things that do clear->add can delay redraw until after the new overlays are added)
+//
+// Each of the lists can contain their own list of sublists if needed,
+// Associated values are also supported and in those cases, the value is what will be added, not the key
+//		(This means no more storing a seperate list of overlays for the object, you can just key it to a value)
+// 		(You can ofcouse also do a sublists of associated values

```

Comment:
It's an American term that the British cannot hope to comprehend, that's why. :^)